bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/wizard/WizardNewProjectCreationPage.java

```diff
@@ -53,14 +56,17 @@
 	// initial value stores
 	private String initialProjectFieldValue;
 
+	private TemplateListSelectionPage templatesPage;
 	// widgets
 	Text projectNameField;
 
 	private Listener nameModifyListener = e -> {
+		if (templatesPage.isCreated()) {
+			templatesPage.setProjectName(projectNameField.getText());
+		}
```

Comment:
The better way to get the project name in the TemplateListSelectionPage is to make use of the eclipse wizard **getPreviousPage()** instance and get the name. Can you explore more on this?